code/game/objects/structures/crates_lockers/closets/coffin.dm

```diff
@@ -6,13 +6,143 @@
 	icon_opened = "coffin_open"
 
 	starting_materials = list(MAT_WOOD = 5*CC_PER_SHEET_MISC)
+	var/mob_lock_type = /datum/locking_category/buckle/closet/coffin
+
 
 /obj/structure/closet/coffin/Destroy()
 	new /obj/item/stack/sheet/wood(loc,3) //This will result in 3 dropped if destroyed, or 5 if deconstructed
+	if (get_locked(mob_lock_type).len) //if someone is strapped in and this gets destroyed make them visible again
+		var/mob/locked = get_locked(mob_lock_type)[1]	
+		locked.alphas["coffin_invis"] = 255
+		locked.handle_alpha()
 	..()
 
 /obj/structure/closet/coffin/update_icon()
 	if(!opened)
 		icon_state = icon_closed
 	else
-		icon_state = icon_opened
\ No newline at end of file
+		icon_state = icon_opened
+
+/datum/locking_category/buckle/closet/coffin
+	flags = LOCKED_SHOULD_LIE
+
+/obj/structure/closet/coffin/attack_hand(mob/user as mob)
+	..()
+	handle_user_visibility()
+
+/obj/structure/closet/coffin/proc/has_locked_mobs()
+	if (!get_locked(mob_lock_type).len)
+		return FALSE
+	var/mob/locked = get_locked(mob_lock_type)[1]
+	if (locked) //no need to try to move if you are strapped in
+		return TRUE
+	else 
+		return FALSE
+
+/obj/structure/closet/coffin/verb/verb_togglebuckle()
+	set src in oview(1)
+	set category = "Object"
+	set name = "Toggle Buckle"
+
+ 	handle_buckle(usr)
+
+/obj/structure/closet/coffin/proc/handle_buckle() //needs src.opened otherwise bugs might occur because closet eats the items when its closed
```

Comment:
Relying on `usr` is ill-advised. Just have `handle_buckle(var/mob/user)` and use `user` rather than `usr`.